### PR TITLE
#708 Added log dirs alter operation

### PR DESCRIPTION
**Pull request by tdavletov**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
Added ability to move partition between log directories

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [x] Manually(please, describe, when necessary) - Tested happy path manually because it's too complex to create itest for happy path (need to change broker configuration before start) 
- [ ] Unit checks
- [x] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings(e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java [hunk 1]
```diff
@@ -41,4 +43,11 @@
   ) {
     return Mono.just(ResponseEntity.ok(clusterService.getAllBrokersLogdirs(clusterName, brokers)));
   }
+
+  @Override
+  public Mono<ResponseEntity<BrokerLogdirUpdateResult>> updateBrokerTopicPartitionLogDir(
+      String clusterName, Integer id, Mono<BrokerLogdirUpdate> brokerLogdir,
+      ServerWebExchange exchange) {
+    return clusterService.updateBrokerLogDir(clusterName, id, brokerLogdir).map(ResponseEntity::ok);
+  }
```
Comment (IldarAlmakaev) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java:52]: Timur, could you please refactor your method not to pass Mono/Flux objects as a function parameter? For example, just using
`brokerLogDir.map(... -> clusterService.updateBrokerLogDir(...))` instead?
I know there're a lot of places where we've done it in such a way.

**Review by IldarAlmakaev — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java (hunk 1 above)
Comment (tdavletov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java:54]: Fixed, please review

**Review by IldarAlmakaev — APPROVED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java [hunk 2]
```diff
@@ -906,6 +908,34 @@ private InternalClusterMetrics calculateClusterMetrics(
...
+    Map<TopicPartitionReplica, String> req = Map.of(
+        new TopicPartitionReplica(b.getTopic(), b.getPartition(), broker),
+        b.getLogDir());
+    return Mono.just(adminMono)
+        .map(admin -> admin.getAdminClient().alterReplicaLogDirs(req))
+        .flatMap(result -> ClusterUtil.toMono(result.all()))
+        .then(Mono.just(result(BrokerLogdirUpdateResult.StatusEnum.OK, null)))
+        .onErrorResume(ApiException.class, (ApiException e) -> {
+          log.error("Error during updating log dirs", e);
+          return Mono.just(result(BrokerLogdirUpdateResult.StatusEnum.ERROR, e.getMessage()));
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:930]: Looks like you'll return 200 even if broker sent error. I think it's better to pass 4xx/5xx status

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java (hunk 2 above)
Comment (tdavletov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java:930]: Fixed, please review

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 3]
```diff
@@ -172,6 +172,35 @@ paths:
                 items:
                   $ref: '#/components/schemas/BrokersLogdirs'
 
+  /api/clusters/{clusterName}/brokers/{id}/logDirs:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:175]: Please do not use camel case in paths

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java (hunk 1 above)
Comment (germanosin) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java:57]: To be precise, it's not always Bad Request, we should provide 5xx errors also, and maybe it's better to throw on Mono level, to catch and provide as other errors

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 3 above)
Comment (tdavletov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:175]: Url is copied from your task description :) But ok, I'll fix it.

Comment (germanosin) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:175]: sorry, my bad)

Comment (tdavletov) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml]: Fixed

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java (hunk 1 above)
Comment (tdavletov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java:53]: Fixed

**Review by germanosin — APPROVED** [review]

**Pull request merged**